performance-tests/bench/dashboard_summarizer/main.cpp

```diff
@@ -0,0 +1,193 @@
+#include "util.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <iostream>
+
+using namespace rapidjson;
+
+namespace {
+
+Value& find_or_create(Value& parent, const std::string& name, Type type, Document::AllocatorType& alloc)
+{
+  Value::MemberIterator pos = parent.FindMember(Value(name.c_str(), name.length()));
+  if (pos == parent.MemberEnd()) {
+    parent.AddMember(Value(name.c_str(), name.length(), alloc).Move(), Value(type).Move(), alloc);
+    pos = parent.FindMember(Value(name.c_str(), name.length()));
+    assert(pos != parent.MemberEnd());
+  }
+  assert(pos->value.GetType() == type);
+  return pos->value;
+}
+
+void transfer_double(Value& out, Document::AllocatorType& alloc, const Value& in, const std::string& in_name, const std::string out_name)
+{
+  double to_write = 0.0;
+  auto pos = in.FindMember(Value(in_name.c_str(), in_name.length()));
+  if (pos != in.MemberEnd() && pos->value.IsDouble()) {
+    to_write = pos->value.GetDouble();
+  }
+  assert(out.GetType() == kObjectType);
+  Value& double_value = find_or_create(out, out_name, kNumberType, alloc);
+  double_value.SetDouble(to_write);
+  assert(out.GetType() == kObjectType);
+}
+
+void transfer_uint64(Value& out, Document::AllocatorType& alloc, const Value& in, const std::string& in_name, const std::string out_name)
+{
+  uint64_t to_write = 0;
+  auto pos = in.FindMember(Value(in_name.c_str(), in_name.length()));
+  if (pos != in.MemberEnd() && pos->value.IsUint64()) {
+    to_write = pos->value.GetUint64();
+  }
+  assert(out.GetType() == kObjectType);
+  Value& double_value = find_or_create(out, out_name, kNumberType, alloc);
+  double_value.SetUint64(to_write);
+  assert(out.GetType() == kObjectType);
+}
+
+void write_statistic(Value& parent_out, Document::AllocatorType& alloc, const std::string& name, const Value& in)
+{
+  Value& out = find_or_create(parent_out, name, kObjectType, alloc);
+  assert(out.GetType() == kObjectType);
+
+  transfer_uint64(out, alloc, in, "sample_count", "count");
+
+  transfer_double(out, alloc, in, "min", "min");
+  transfer_double(out, alloc, in, "max", "max");
+  transfer_double(out, alloc, in, "mean", "mean");
+  transfer_double(out, alloc, in, "stdev", "stdev");
+  transfer_double(out, alloc, in, "median", "median");
+  transfer_double(out, alloc, in, "madev", "madev");
+
+  transfer_uint64(out, alloc, in, "median_sample_overflow", "overflow");
+}
+
+}
+
+int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
+{
+  std::vector<std::string> file_names;
+  std::vector<std::shared_ptr<std::ifstream> > files;
```

Comment:
```suggestion
  std::vector<std::shared_ptr<std::ifstream>> files;
```
Since we're assuming C++11 here